### ISSUE TITLE
Fails to parse `ipfs-core`

### ISSUE THREAD

**Issue by patrickmichalina** [meriyah/meriyah#165]
When importing `ipfs-core` in fusebox, receive the following error:

```
Error while parsing module /app/node_modules/ipfs-core/src/components/files/chmod.js
        ' SyntaxError: [212:25]: Unexpected token: 'async'
    at report (/app/node_modules/meriyah/dist/meriyah.umd.js:189:13)
```

line 212 is `yield async function ()`

```
dagBuilder: async function * (source, block, opts) {
            for await (const entry of source) {
              yield async function () {
                const cid = await persist(entry.content.serialize(), block, opts)

                return {
                  cid,
                  path: entry.path,
                  unixfs: UnixFS.unmarshal(entry.content.Data),
                  node: entry.content
                }
              }
            }
          }
```

I am not sure if this is a fusebox issue @nchanged

**Comment by KFlash** [comment]
cc @3cp I did an online edit to try to fix this, but need to verify with tests. There was an issue in 'token.ts'

**Comment by 3cp** [comment]
Released the fix in v4.0.0, note there is breaking change on estree node name, not sure if that affects fusebox.
